index.js

```diff
@@ -1034,6 +1040,54 @@ const createGoBom = async (includeBomSerialNumber, path, options, callback) => {
       },
       callback
     );
+    return;
+  }
+
+  // If USE_GOSUM is false, generate BOM components using go.mod.
+  if (gosumFiles.length) {
+    goSumReader = [];
+    for (let i in gosumFiles) {
+      const f = gosumFiles[i];
+      if (DEBUG_MODE) {
+        console.log(`Parsing ${f}`);
+      }
+      goSumReader.push(fs.readFileSync(f, { encoding: "utf-8" }));
```

Comment:
Instead of holding the raw text perhaps here we build up an object with key being the full name and value containing the hash. This can be passed to parseGoModData directly.